internal/cluster/distribution/eks/service.go

```diff
@@ -61,29 +66,48 @@ type NodePoolUpdateDrainOptions struct {
 	PodSelector string `mapstructure:"podSelector"`
 }
 
+type SecretStore interface {
+	Get(orgnaizationID uint, secretID string) (*secret.SecretItemResponse, error)
+	GetByName(orgnaizationID uint, secretID string) (*secret.SecretItemResponse, error)
+	Store(organizationID uint, request *secret.CreateSecretRequest) (string, error)
+	Delete(organizationID uint, secretID string) error
+	Update(organizationID uint, secretID string, request *secret.CreateSecretRequest) error
+}
```

Comment:
Do you need all of these methods here? In Go, interfaces are owned by consumers so that they can define what they need.

Furthermore, if you inject the aws session factory into the manager, you don't need the secret store at all.